network/src/primary.rs

```diff
@@ -35,16 +38,16 @@ fn default_executor() -> BoundedExecutor {
 
 impl Default for PrimaryNetwork {
     fn default() -> Self {
-        let retry_config = RetryConfig {
-            // Retry for forever
-            retrying_max_elapsed_time: None,
-            ..Default::default()
-        };
+        let backoff = Backoff::new(
+            u32::MAX,
```

Comment:
>can we find a way to better convey intent? 

100%. I will circle back on this once cancellation is implemented. I think the approach you outlined makes sense.

>before this the returned JoinHandle was always a straightforward way to cancel the overall retried task, and reclaim all of its resources. Now, with a task being sent to the scheduler, how would that work?

It is still a work-in-progress, but you note a key point here. The previous implementation mapped perfectly to a `JoinHandle, indeed we had one retrier task managing each retry attempt. A retry attempt can be:

- in the execution queue
- awaiting a semaphore permit
- running
- expired

A `JoinHandle` only covers a subset of this domain, so I think there's a good reason to introduce a `RetryHandle` , or perhaps `JoinRetryHandle`, that abstracts the entire lifecycle of a retry attempt. That handle would have an`abort` implementation accounting for the different states listed above. The goal is to make resource reclamation possible and fit with the `CancelOnDrop` approach used in the primary/work implementations. 

As a bonus, this also paves the way to enabling callers to await completion of a retry attempt, instead of always resolving after the first try regardless of the outcome. I know this is not a priority, since that's not how retries are used in our case.